pcsx2/gui/Dialogs/GSDumpDialog.cpp

```diff
@@ -0,0 +1,776 @@
+	prim_infos[7].Printf("CTXT = %s", GsCTXTNames[((prim >> 9) & 1)]);
+	prim_infos[8].Printf("FIX = %s", GsFIXNames[((prim >> 10) & 1)]);
+
+	for (auto& el : prim_infos)
+		m_gif_packet->AppendItem(id, el);
+}
+
+void Dialogs::GSDumpDialog::CheckDebug(wxCommandEvent& event)
+{
+	if (m_debug_mode->GetValue())
+		GenPacketList(m_dump_packets);
+	else
+	{
+		m_gif_list->DeleteAllItems();
+		m_gif_packet->DeleteAllItems();
+		m_gif_list->Refresh();
+	}
+}
+
+Dialogs::GSDumpDialog::GSThread::GSThread(GSDumpDialog* dlg)
+	: pxThread("GSDump")
+	, m_root_window(dlg)
+{
+}
+
+Dialogs::GSDumpDialog::GSThread::~GSThread()
+{
+	try
+	{
+		pxThread::Cancel();
+	}
+	DESTRUCTOR_CATCHALL
+}
+
+void Dialogs::GSDumpDialog::GSThread::OnStop()
+{
+	m_root_window->m_debug_mode->Disable();
+	m_root_window->m_start->Disable();
+	m_root_window->m_step->Disable();
+	m_root_window->m_selection->Disable();
+	m_root_window->m_vsync->Disable();
+	m_root_window->m_gif_list->DeleteAllItems();
+	m_root_window->m_gif_packet->DeleteAllItems();
+	m_root_window->m_gif_list->Refresh();
+	m_root_window->m_button_events.clear();
+	m_dump_file->Close();
+}
+
+void Dialogs::GSDumpDialog::GSThread::ExecuteTaskInThread()
+{
+	u32 crc = 0, ss = 0;
+	// XXX: check the numbers are correct
+	const int renderer_override = m_root_window->m_renderer_overrides->GetSelection();
+	char regs[8192];
+
+	m_dump_file->Read(&crc, 4);
+	m_dump_file->Read(&ss, 4);
+
+
+	std::unique_ptr<char[]> state_data(new char[ss]);
+	m_dump_file->Read(state_data.get(), ss);
+	m_dump_file->Read(&regs, 8192);
+
+	freezeData fd = {(int)ss, (s8*)state_data.get()};
+	m_root_window->m_dump_packets.clear();
+
+	while (	m_dump_file->Tell() < m_dump_file->Length())
+	{
+		GSType id = Transfer;
+		m_dump_file->Read(&id, 1);
+		switch (id)
+		{
+			case Transfer:
+			{
+				GSTransferPath id_transfer;
+				m_dump_file->Read(&id_transfer, 1);
+				s32 size = 0;
+				m_dump_file->Read(&size, 4);
+				std::unique_ptr<char[]> transfer_data(new char[size]);
+				m_dump_file->Read(transfer_data.get(), size);
+				GSData data = {id, std::move(transfer_data), size, id_transfer};
+				m_root_window->m_dump_packets.emplace_back(data);
+				break;
+			}
+			case VSync:
+			{
+				std::unique_ptr<char[]> vsync(new char[1]);
+				m_dump_file->Read(vsync.get(), 1);
+				GSData data = {id, std::move(vsync), 1, Dummy};
+				m_root_window->m_dump_packets.emplace_back(data);
+				break;
+			}
+			case ReadFIFO2:
+			{
+				std::unique_ptr<char[]> fifo(new char[4]);
+				m_dump_file->Read(fifo.get(), 4);
+				GSData data = {id, std::move(fifo), 4, Dummy};
+				m_root_window->m_dump_packets.emplace_back(data);
+				break;
+			}
+			case Registers:
+			{
+				std::unique_ptr<char[]> regs_tmp(new char[8192]);
+				m_dump_file->Read(regs_tmp.get(), 8192);
+				GSData data = {id, std::move(regs_tmp), 8192, Dummy};
+				m_root_window->m_dump_packets.emplace_back(data);
+				break;
+			}
+		}
```

Comment:
Repaste from discord:

Give GSData a constructor that takes the id and the file pointer. You can then move the switch statement inside the constructor and manipulate the unique_ptr member variable directly. You would no longer need std::move and would simply this section to a single emplace_back call.